reference/conanfile/methods/requirements.rst

```diff
@@ -257,3 +264,12 @@ Each kind of requires sets some additional traits by default on top of the ones
  * ``build_requires``:  ``headers=False``, ``libs=False``, ``build=True``, ``visible=False``
  * ``tool_requires``: ``headers=False``, ``libs=False``, ``build=True``, ``run=True``, ``visible=False``
  * ``test_requires``: ``headers=True``, ``libs=True``, ``build=False``, ``visible=False``, ``test=True``
+
+For example, taking all the logic shown into account, this means that if you have
+``self.requires("mypkg/1.0", visible=False)`` and ``mypkg/1.0`` has ``package_type="shared-library"``,
+the effective traits for that ``requires`` will be:
+
+ * Inferred from ``mypkg``'s ``package_type``: ``run=True``
+ * By the ``requires`` kind: ``build=False``
+ * Overridden by the user: ``visible=False``
+ * By default: ``headers=True``, ``libs=True``
```

Comment:
Note that both ``run=True`` and ``visible=False`` are kind of incompatible. Unless the shared-library isn't really used by the consumer, for example, lets say that the consumer is good with just a headers with some templates that don't have any linkage needs to the shared library, this will be problematic, as the shared-library runtime requirements will not be propagated downstream, and it will fail to run.